python/hail/methods/statgen.py

```diff
@@ -126,3 +129,191 @@ def ld_matrix(dataset, force_local=False):
 
     jldm = Env.hail().methods.LDMatrix.apply(dataset._jvds, force_local)
     return LDMatrix(jldm)
+
+
+@handle_py4j
+@require_biallelic
+@typecheck(dataset=MatrixTable,
+           k=integral,
+           compute_loadings=bool,
+           as_array=bool)
+def hwe_normalized_pca(dataset, k=10, compute_loadings=False, as_array=False):
+    """Run principal component analysis (PCA) on the Hardy-Weinberg-normalized call matrix.
+
+    Examples
+    --------
+
+    >>> eigenvalues, scores, loadings = methods.hwe_normalized_pca(dataset, k=15)
+
+    Notes
+    -----
+    Variants that are all homozygous reference or all homozygous variant are removed before evaluation.
+
+    Parameters
+    ----------
+    dataset : :class:`MatrixTable`
+        Dataset.
+    k : :obj:`int`
+        Number of principal components.
+    compute_loadings : :obj:`bool`
+        If ``True``, compute row loadings.
+    as_array : :obj:`bool`
+        If ``True``, return scores and loadings as an array field. If ``False``, return
+        one field per element (`PC1`, `PC2`, ... `PCk`).
+
+    Returns
+    -------
+    (:obj:`list` of :obj:`float`, :class:`Table`, :class:`Table`)
+        List of eigenvalues, table with column scores, table with row loadings.
+    """
+
+    dataset = dataset.annotate_rows(AC=agg.sum(dataset.GT.num_alt_alleles()),
+                                    n_called=agg.count_where(functions.is_defined(dataset.GT)))
+    dataset = dataset.filter_rows((dataset.AC > 0) & (dataset.AC < 2 * dataset.n_called)).persist()
+
+    n_variants = dataset.count_rows()
+    if n_variants == 0:
+        raise FatalError(
+            "Cannot run PCA: found 0 variants after filtering out monomorphic sites.")
+    info("Running PCA using {} variants.".format(n_variants))
+
+    # FIXME: use bind
```

Comment:
Do you want to fix this now that bind has gone in?

Reply:
yup, I do. Pretty much every PR I've made in the last week breaks every other ☹️